src/com/ka/spreadsheet/diff/UnifiedDiffSpreadSheetDiffCallback.java

```diff
@@ -0,0 +1,168 @@
+package com.ka.spreadsheet.diff;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.ka.spreadsheet.diff.SpreadSheetUtils.CELL_INTERNAL_TO_USER;
+
+// Output format, inspired by the traditional Unified Diff format (but using cell references instead of line numbers):
+//      +--- column 1
+//      V
+//        --- file_1_name!sheet_1_name
+//        +++ file_2_name!sheet_1_name
+//        @@ cell_1_ref,cell_n_ref cell_1_ref,cell_n_ref @@
+//        -old_cell_1_data
+//        -...
+//        -old_cell_n_data
+//        +new_cell_1_data
+//        +...
+//        +new_cell_n_data
+//        @@@ ...
+//        ...
+//        --- file_1_name!sheet_2_name
+//        +++ file_2_name!sheet_2_name
+//        ...
+// Each cell data line is always present, even if the data is empty (thus just "-" or "+).
+public class UnifiedDiffSpreadSheetDiffCallback implements SpreadSheetDiffCallback {
+
+  private final String lineSeparator = System.getProperty("line.separator");
+  private String file1;
+  private String file2;
+  private CellPos previousCell = null;
```

Comment:
can initialize this one also to some value which will be < all real values

Reply:
Creating a `CellPos` from scratch requires creating an `ISheet` and a `CellValue`, and maybe more.  That was more trouble than I wanted to go to for this.  This is specifically to support the diff/callback contract assertions, so a fake `CellPos` would have to satisfy `CellPos.compareCellPositions()`.  It seemed a whole lot easier to use `null` to represent "no cells observed yet".

Reply:
Sounds good